Modules/gcmodule.c

```diff
@@ -1257,7 +1257,8 @@ static PyObject *
 gc_enable_impl(PyObject *module)
 /*[clinic end generated code: output=45a427e9dce9155c input=81ac4940ca579707]*/
 {
-    _PyRuntime.gc.enabled = 1;
+    PyInterpreterState *interp = _PyInterpreterState_GET_UNSAFE();
```

Comment:
Would it be possible to use a "struct _gc_runtime_state *state" variable instead?

```
struct _gc_runtime_state *state = &_PyInterpreterState_GET_UNSAFE()->gc;
```

If the line is too lone, maybe add a GET_STATE() macro which returns ``&_PyInterpreterState_GET_UNSAFE()->gc``.

Same comment for the whole file. I would prefer to avoid introducing "PyInterpreterState *interp = _PyInterpreterState_GET_UNSAFE(); " lines if it's only used to get the GC state.